pkg/container-collection/options.go

```diff
@@ -585,6 +581,34 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigEnrichment enriches container using provided OCI config
+func WithOCIConfigEnrichment() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
+			if container.OciConfig != nil {
+				// skip for pause container
+				if ct, ok := container.OciConfig.Annotations[crioContainerType]; ok && ct == "sandbox" {
+					return false
+				}
```

Comment:
I don't think we should skip pause containers, as local-gadget users may also want to debug those. In fact, the ContainerRuntime enricher doesn't drop them.

Reply:
Got it. I realized that pause containers were being printed with `list-containers --watch` I feel like since we aren't listing them with non-watched listing we should skip them during watch as well. What do you think?

Reply:
Mmmm I remember to have seen the pause containers only for Docker on both cases, watched and non-watched. What runtime were you using when you saw this behaviour?

Reply:
CRI-O on kvm driver. I am bit confused regrading pause container and pod sandbox because the notifications I was seeing are related to pod sandbox. We don't need to show them to users. Right?

Reply:
I think I have an idea what confused me. We aren't listing the pod sandboxes in the [CRI client](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/cri/cri.go#L80) so they are dropped during enrichment even for `contrainerd`. If we decided to handle them here we should  adopt the cri client to handle them as well.

Reply:
Ahh, you are right. We should use [`ListPodSandbox`](https://pkg.go.dev/k8s.io/cri-api@v0.20.6/pkg/apis/runtime/v1alpha2#RuntimeServiceClient.ListPodSandbox) to get the pause containers. While using the Docker API Engine, we get all of them with [the same API](https://pkg.go.dev/github.com/docker/docker@v20.10.8+incompatible/client#Client.ContainerList).

I propose to continue ignoring the pause containers for CRI-O and containerd in this PR, and manage that [issue](https://github.com/inspektor-gadget/inspektor-gadget/issues/1095) in another one.